airbyte-integrations/connectors/source-rss/source_rss/source.py

```diff
@@ -0,0 +1,156 @@
+
+    # since we only have one response for the stream, we should only return records newer than the initial state object if incremental
+    def parse_response(self, response: requests.Response, stream_state: MutableMapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        feed = feedparser.parse(response.text)
+
+        try:
+            initial_state_date = parse(stream_state["published"])
+        except:
+            initial_state_date = None
+
+        # go through in reverse order which helps the state comparisons
+        all_item_mappings = [convert_item_to_mapping(item) for item in feed.entries[::-1]]
+
+        # will only filter if we have a state object, so it's incremental
+        yield from [item for item in all_item_mappings if is_newer(item, initial_state_date)]
+
+
+# Basic incremental stream
+class IncrementalRssStream(RssStream, ABC):
+    # no reason to checkpoint if it's reading individual files without pagination
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        return "published"
+
+    # this will fail if the dates aren't parseable, but that means incremental isn't possible anyway for that feed
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        try:
+            latest_record_date = parse(latest_record["published"])
+            latest_record_state = {"published": latest_record["published"]}
+        except:
+            latest_record_date = None
+
+        try:
+            current_record_date = parse(current_stream_state["published"])
+        except:
+            current_record_date = None
+
+        if latest_record_date and current_record_date:
+            if latest_record_date > current_record_date:
+                return latest_record_state
+            else:
+                return current_stream_state
+        if latest_record_date:
+            return latest_record_state
+        if current_record_date:
+            return current_stream_state
+        else:
+            return {}
+
+
+class Items(IncrementalRssStream):
+    def __init__(self, url: str):
+        super().__init__()
+        self.url = url
+
+    primary_key = None
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return self.url
+
+
+# Source
+class SourceRss(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        try:
+            resp = requests.get(config.get("url"))
+            status = resp.status_code
+            if status == 200:
+                return True, None
+            else:
+                return False, f"Unable to connect to RSS Feed (received status code: {status})"
+        except Exception as e:
+            return False, e
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
```

Comment:
@jrhizor would have been great to see a stream generator based on multiple URL entry, but happy for someone else to have a go at that next :)

Reply:
I thought about allowing multiple URLs but it seemed easier to manage separate feeds in separate sources because they can fail independently or could have their own custom configuration if (in the future) the connector supported manually specifying other  `<item>` sub-elements you wanted to capture. The main benefit would be to reduce the overhead on the system, but Airbyte Cloud doesn't charge in a way that penalizes based on the number of connectors or container launching overhead.

Another good extension in the future would be to add support for Atom feeds (should be really easy considering that `feedburner` supports both).